stubs/colorful/colorful/colors.pyi

```diff
@@ -0,0 +1,4 @@
+def parse_colors(path: str) -> dict[str, str]: ...
+def parse_rgb_txt_file(path: str) -> dict[str, str]: ...
+def parse_json_color_file(path: str) -> dict[str, str]: ...
+def sanitize_color_palette(colorpalette: dict[str, str | tuple[int, int, int]]) -> dict[str, tuple[int, int, int]]: ...
```

Comment:
Here it's safer to use `SupportsItems` from `_typeshed`:

```suggestion
def sanitize_color_palette(colorpalette: SupportsItems[str, str | tuple[int, int, int]]) -> dict[str, tuple[int, int, int]]: ...
```

(Also needs `from _typeshed import SupportsItems` at the top of the file.)

This prevents variance issues. For example, this would show an error, because `dict[str, str]` is not compatible with `dict[str, str | tuple[int, int, int]]`:

```python
d = {"red": "#ff0000"}
sanitize_color_palette(d)
```

It's also better to use tight protocols in argument annotations to enable duck typing.

Reply:
thanks :sparkles: i was not aware that this is how that works 
while playing around with it i also seen Mapping 

```
from typing import Mapping

def fun(value: Mapping[str, str | tuple[int, int, int]]):
    print(value)


x = {"red": "#ff0000"}
fun(x)

y = {"red": (2,3,4)}
fun(y)
```

Is there a reason to use SupportsItems over Mapping?

Reply:
`Mapping` is not a protocol and it's also quite broad. We currently overuse `Mapping` in typeshed, unfortunately.